src/Components/PaperEditor/ProgrammeScriptContainer/index.js

```diff
@@ -281,12 +281,64 @@ const ProgrammeScriptContainer = (props) => {
           };
           newElement.words.push(newWord);
         });
+
+        elementsClone.splice(insertElementIndex, 0, newElement);
+        const updatedElements = updateWordTimingsAfterInsert(elementsClone, insertElementIndex);
+        setElements(updatedElements);
+        setResetPreview(true);
+        handleSaveProgrammeScript(updatedElements);
+
+      } else {
+        const wordsArray = divideWordsSelectionsIntoParagraphs(result.words);
```

Comment:
```suggestion
        const paragraphSelections = divideWordsSelectionsIntoParagraphs(result.words);
```
Suggestion for name change, such as `wordsSelections` ? `paragraphSelections` ?